sdk/core/azure-core/src/main/java/com/azure/core/credential/SimpleTokenCache.java

```diff
@@ -77,16 +75,29 @@ public class SimpleTokenCache {
     private final Supplier<Mono<AccessToken>> tokenSupplier;
     private final Predicate<AccessToken> shouldRefresh;
 
+    // The delay after a refresh to attempt another token refresh
+    private final Duration refreshDelay;
+    private final String refreshDelayString;
+
     /**
      * Creates an instance of RefreshableTokenCredential with default scheme "Bearer".
      *
      * @param tokenSupplier a method to get a new token
      */
     public SimpleTokenCache(Supplier<Mono<AccessToken>> tokenSupplier) {
+        this(tokenSupplier, Duration.ofSeconds(30));
+
+    }
+
+    SimpleTokenCache(Supplier<Mono<AccessToken>> tokenSupplier, Duration refreshDelay) {
```

Comment:
We should add a comment that this constructor is only for testing purposes.

Reply:
in a follow up PR.